x-pack/plugins/canvas/server/feature.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { DEFAULT_APP_CATEGORIES } from '../../../../src/core/server';
+import { KibanaFeatureConfig } from '../../features/common';
+import { ReportingSetup } from '../../reporting/server';
+
+/*
+ * Register Canvas as a Kibana feature,
+ * with Reporting sub-feature integration (if enabled)
+ */
+export function getCanvasFeature(plugins: { reporting?: ReportingSetup }): KibanaFeatureConfig {
+  const includeReporting = plugins.reporting && plugins.reporting.usesUiCapabilities();
+
+  return {
+    id: 'canvas',
+    name: 'Canvas',
+    order: 300,
+    category: DEFAULT_APP_CATEGORIES.kibana,
+    app: ['canvas', 'kibana'],
+    management: {
+      ...(includeReporting ? { insightsAndAlerting: ['reporting'] } : {}),
+    },
+    catalogue: ['canvas'],
+    privileges: {
+      all: {
+        app: ['canvas', 'kibana'],
+        catalogue: ['canvas'],
+        savedObject: {
+          all: ['canvas-workpad', 'canvas-element'],
+          read: ['index-pattern'],
+        },
+        ui: ['save', 'show'],
+      },
+      read: {
+        app: ['canvas', 'kibana'],
+        catalogue: ['canvas'],
+        savedObject: {
+          all: [],
+          read: ['index-pattern', 'canvas-workpad', 'canvas-element'],
+        },
+        ui: ['show'],
+      },
+    },
+    subFeatures: [
```

Comment:
Interesting -- I was not able to reproduce that. I started with a Trial license, and downgraded to Basic. With the Basic license, I created a new role - and during that I verified that I could not create custom sub-feature privileges. Then I uploaded a Platinum license. I went back to the role that I created, and found that I can customize the sub-feature privileges.

Reply:
@tsullivan it didn't happen if I was on a trial when Kibana started.  It was only if I started up on Basic, upgraded, and then checked it.  Stopping and starting Kibana would resolve it since it would then be starting up with the trial.

Reply:
I haven't tried this, but it seems like an issue with the Roles UI. I made a note to myself to follow up on this later and file a bug with the details.